storage/client_raft_test.go

```diff
@@ -651,3 +652,30 @@ func TestReplicateAddAndRemove(t *testing.T) {
 		verify([]int64{39, 0, 39, 39})
 	}
 }
+
+// TestRaftHeartbeats verifies that coalesced heartbeats are correctly
+// suppressing elections in an idle cluster.
+func TestRaftHeartbeats(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	mtc := startMultiTestContext(t, 3)
+	defer mtc.Stop()
+	mtc.replicateRange(1, 0, 1, 2)
+
+	// Capture the initial term and state.
+	status := mtc.stores[0].RaftStatus(1)
+	initialTerm := status.Term
+	if status.SoftState.RaftState != raft.StateLeader {
+		t.Errorf("expected node 0 to initially be leader but was %s", status.SoftState.RaftState)
+	}
+
+	// Wait for several ticks to elapse.
+	time.Sleep(5 * mtc.makeContext().RaftTickInterval)
```

Comment:
Here, the event we want to wait for is multiraft.state.fanoutHeartbeatResponse, but this may be the only current test that would be interested in that event. Other heartbeat-related tests would be interested in when fanoutHeartbeat decides *not* to send a heartbeat, and others are interested in the number of messages that are actually sent on the wire. We'd have to scatter these events all over the code base, with tight coupling between tests and code. It might be worth it for determinism, but it won't be easy.